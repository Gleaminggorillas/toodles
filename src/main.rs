use crate::app::App;

pub mod app;
pub mod event;
pub mod ui;

fn main() -> color_eyre::Result<()> {
    color_eyre::install()?;
    let terminal = ratatui::init();
    let result = App::new().run(terminal);
    ratatui::restore();
    result
}

// fn run(mut terminal: DefaultTerminal) -> Result<()> {
//     loop {
//         terminal.draw(draw)?;
//         if let Event::Key(key) = event::read()? {
//             if key.kind == KeyEventKind::Press && key.code == KeyCode::Char('q') {
//                 break Ok(());
//             }
//         }
//     }
// }

#[cfg(test)]
mod tests {
    use super::*;
    use insta::assert_snapshot;
    use ratatui::widgets::Padding;
    use ratatui::{Terminal, backend::TestBackend};

    #[test]
    fn add() {
        assert_eq!(1 + 1, 2);
    }

    #[test]
    fn test_render_app() {
        // Given the app is running
        let app = App::default();
        // When the application is open
        let mut terminal = Terminal::new(TestBackend::new(80, 20)).unwrap();
        terminal
            .draw(|frame| frame.render_widget(&app, frame.area()))
            .unwrap();
        // Then the snapshot is accurate
        assert_snapshot!(terminal.backend());
    }
}
