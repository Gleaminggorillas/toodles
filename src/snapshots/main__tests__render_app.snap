---
source: src/main.rs
expression: terminal.backend()
---
"╭───────────────────────────────{{project-name}}───────────────────────────────╮"
"│                            This is a tui template.                           │"
"│                 Press `Esc`, `Ctrl-C` or `q` to stop running.                │"
"│   Press left and right to increment and decrement the counter respectively.  │"
"│                                  Counter: 0                                  │"
"│                                                                              │"
"│                                                                              │"
"│                                                                              │"
"│                                                                              │"
"│                                                                              │"
"│                                                                              │"
"│                                                                              │"
"│                                                                              │"
"│                                                                              │"
"│                                                                              │"
"│                                                                              │"
"│                                                                              │"
"│                                                                              │"
"│                                                                              │"
"╰──────────────────────────────────────────────────────────────────────────────╯"
